user_guide_src/source/libraries/validation/043.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+$validation = \Config\Services::validation();
+$validation->setRules([
+    'username' => 'required',
+    'password' => 'required|min_length[10]',
+]);
+
+$data = [
+    'username'   => 'john',
+    'password'   => 'BPi-$Swu7U5lm$dX',
+    'csrf_token' => '8b9218a55906f9dcc1dc263dce7f005a',
+];
+
+if ($validation->run($data)) {
+    $validatedData = $validation->getValidated();
```

Comment:
This example is fine, but I would like to ask you to add an example that will cover controller validation (with the [$this->validate()](https://github.com/codeigniter4/CodeIgniter4/blob/develop/user_guide_src/source/incoming/controllers.rst#this-validate) method). This is the place where this method is the most valuable.

Reply:
I have a plan to add a method to get validated data in `Controller`, and update all sample code with `$this->validate()`.

Reply:
I don't think need a separate method in the controller for this, though. 
Since we can get the errors via `$this->validator->getErrors()`, I see no problem with getting validated data the same way, via `$this->validator->getValidated()`.

Reply:
Oh, I see. Thank you for opinion.

Reply:
See #7425